packages/svelte/src/internal/client/reactivity/batch.js

```diff
@@ -857,6 +863,12 @@ export function schedule_effect(signal) {
 		if ((flags & (ROOT_EFFECT | BRANCH_EFFECT)) !== 0) {
 			if ((flags & CLEAN) === 0) return;
 			effect.f ^= CLEAN;
+
+			// Track branches toggled during fork execution so we can restore
+			// their CLEAN flag after flush
+			if (current_batch !== null && current_batch.is_fork) {
+				(current_batch.toggled_branches ??= new Set()).add(effect);
+			}
```

Comment:
I think it makes sense...there's only one thing that it kinda irks me here: we are restoring ALL the branches to `CLEAN` but wouldn't this lead to over-running?